yt_dlp/networking/_urllib.py

```diff
@@ -54,44 +55,10 @@
 def _create_http_connection(http_class, source_address, *args, **kwargs):
     hc = http_class(*args, **kwargs)
 
+    if hasattr(hc, '_create_connection'):
+        hc._create_connection = create_connection
```

Comment:
I'm at a loss at what this does, can you explain? Why do we always have to overwrite it here?

Reply:
[_create_conn](https://github.com/python/cpython/blob/3.12/Lib/http/client.py#L875)

Reply:
thx